Source/WTF/wtf/Assertions.h

```diff
@@ -541,6 +541,19 @@ constexpr bool assertionFailureDueToUnreachableCode = false;
 #endif
 #endif
 
+#ifdef __cplusplus
+namespace WTF {
+template <typename T>
+struct AlwaysFalse {
+    static constexpr bool value = false;
+};
+} // namespace WTF
+#define STATIC_ASSERT_NOT_REACHED_WITH_DEPENDENT_TYPE(T) do { \
```

Comment:
I think you could just name this STATIC_ASSERT_NOT_REACHED. The fact that the programmer is required to pass it a dependent type is a detail of the limitations of C++. It's not part of the intention the programmer is trying to express.

Reply:
As discussed, I've now split it in _FOR_TYPE and _FOR_VALUE instead (necessary due to C++ limitations, but it also has a beneficial effect of showing the type/value in compiler errors).
Let me know if you can think of better names.